dev-java/oracle-jdk-bin/oracle-jdk-bin-9_pre149.ebuild

```diff
@@ -0,0 +1,308 @@
+#
+# * We still need to work out the exact AWT and JavaFX dependencies
+#   under MacOS. It doesn't appear to use many, if any, of the
+#   dependencies below.
+#
+RDEPEND="!x64-macos? (
+		!headless-awt? (
+			x11-libs/libX11
+			x11-libs/libXext
+			x11-libs/libXi
+			x11-libs/libXrender
+			x11-libs/libXtst
+		)
+		javafx? (
+			dev-libs/glib:2
+			dev-libs/libxml2:2
+			dev-libs/libxslt
+			media-libs/freetype:2
+			x11-libs/cairo
+			x11-libs/gtk+:2
+			x11-libs/libX11
+			x11-libs/libXtst
+			x11-libs/libXxf86vm
+			x11-libs/pango
+			virtual/opengl
+		)
+	)
+	alsa? ( media-libs/alsa-lib )
+	cups? ( net-print/cups )
+	doc? ( dev-java/java-sdk-docs:${SLOT} )
+	fontconfig? ( media-libs/fontconfig:1.0 )
+	!prefix? ( sys-libs/glibc:* )
+	selinux? ( sec-policy/selinux-java )"
+
+# A PaX header isn't created by scanelf so depend on paxctl to avoid
+# fallback marking. See bug #427642.
+DEPEND="app-arch/zip
+	pax_kernel? ( sys-apps/paxctl )"
+
+S="${WORKDIR}/jdk"
+
+check_tarballs_available() {
+	local uri=$1; shift
+	local dl= unavailable=
+	for dl in "${@}" ; do
+		[[ ! -f "${DISTDIR}/${dl}" ]] && unavailable+=" ${dl}"
+	done
+
+	if [[ -n "${unavailable}" ]] ; then
+		if [[ -z ${_check_tarballs_available_once} ]] ; then
+			einfo
+			einfo "Oracle requires you to download the needed files manually after"
+			einfo "accepting their license through a javascript capable web browser."
+			einfo
+			_check_tarballs_available_once=1
+		fi
+		einfo "Download the following files:"
+		for dl in ${unavailable}; do
+			einfo "  ${dl}"
+		done
+		einfo "at '${uri}'"
+		einfo "and move them to '${DISTDIR}'"
+		einfo
+		einfo "If the above mentioned urls do not point to the correct version anymore,"
+		einfo "please download the files from Oracle's java download archive:"
+		einfo
+		einfo "   https://jdk9.java.net/download/"
+		einfo
+	fi
+}
+
+pkg_nofetch() {
+	local distfiles=( $(eval "echo \${$(echo AT_${ARCH/-/_})}") )
+	check_tarballs_available "${JDK_URI}" "${distfiles[@]}"
+
+}
+
+src_unpack() {
+	if use x64-macos ; then
+		pushd "${T}" > /dev/null
+		mkdir dmgmount
+		hdiutil attach "${DISTDIR}"/jdk-${MY_PV}-macosx-x64.dmg \
+			-mountpoint "${T}"/dmgmount
+		local update=$(get_version_component_range 4)
+		[[ ${#update} == 1 ]] && update="0${update}"
+		xar -xf dmgmount/JDK\ $(get_version_component_range 2)\ Update\ ${update}.pkg
+		hdiutil detach "${T}"/dmgmount
+		zcat jdk1${MY_PV%u*}0${update}.pkg/Payload | cpio -idv
+		mv Contents/Home "${WORKDIR}"/jdk${MY_PV}
+		popd > /dev/null
+	else
+		default
+	fi
+
+	# Upstream is changing their versioning scheme every release around 1.8.0.*;
+	# to stop having to change it over and over again, just wildcard match and
+	# live a happy life instead of trying to get this new jdk1.8.0_05 to work.
+	mv "${WORKDIR}"/jdk* "${S}" || die
+}
+
+src_prepare() {
+	default
+
+	if [[ -n ${JAVA_PKG_STRICT} ]] ; then
+		# Mark this binary early to run it now.
+		pax-mark Cm ./bin/javap
```

Comment:
Since the pax line is to enable the ability to do a QA sanity check on the following line, I think it makes sense for it to be in src_prepare as opposed to install (as it's a pre-install sanity check)